### ISSUE TITLE
Apply a role="region" to datagrid filter pop-up container along with a meaningful label.

### ISSUE THREAD

**Issue by lf-collins** [vmware/clarity#5346]
## Describe the accessibility issue

We got a report from AccessLevel where they note that role="region" should be applied to datagrid filter pop-up container along with a meaningful label.

## WCAG Guideline

[Issue - WCAG20]
The datagrid filter pop-ups do not have an appropriate role and label to denote the boundaries of the pop-up.
[Impact]
Screen reader users will have difficulty identifying where the pop-up begins and ends, as well as the purpose of the pop-up.
[Recommendation]
Apply a role="region" to the expanded pop-up container along with a meaningful label provided using the aria-label or aria-labelledby attribute.

## How to reproduce

https://clarity.design/documentation/datagrid/custom-filtering
<img width="878" alt="Screen Shot 2020-12-02 at 3 33 37 PM" src="https://user-images.githubusercontent.com/26111301/100843053-be83d300-34b4-11eb-80d1-ade080e76291.png">

## Versions

**App**

- Angular: [e.g. 6]
- Node: [e.g. 8.10.0]
- Clarity: [e.g. 0.12.5]

**Device:**

- Type: [e.g. MacBook]
- Screen Reader: [e.g. VoiceOver, NVDA]

For screen readers, we primarily support the following combinations, and other combinations will be considered on a case by case basis.

- NVDA + Firefox (Windows)
- VoiceOver + Safari (Apple)

## Additional notes

_Add any other notes about the problem here._

**Comment by gnomeontherun** [comment]
@scroniser is this valid or is it captured already in our list?

**Comment by scroniser** [comment]
@gnomeontherun, this is already on [our list](http://a11y.vmware.com/reporting/viewInstance.php?&table=clarityangular_v4_092420&id=216).
